### PR TITLE
code optimization

### PR DESCRIPTION
**Pull request by Liaojinghui**
This pr does the following three things:

1. add `hashOrPubkey` exception message.
2. remove else statement since it is not working.
3. remove unnecessary variable `hashes_for_verifying`.

### CODE REVIEW

**Review by shargon — DISMISSED** [review]

**Review by shargon — COMMENTED** [review]
@Liaojinghui The coverage is decreased :)

**Comment by Liaojinghui** [comment]
> @Liaojinghui The coverage is decreased :)

I started to hate those rules, but i know we have to. I will check and try to fix it.

**Comment by Liaojinghui** [comment]
@shargon

**Review by shargon — APPROVED** [review]

File: tests/Neo.UnitTests/SmartContract/UT_InteropService.cs [hunk 1]
```diff
@@ -300,6 +300,20 @@ public void TestRuntime_CheckWitness()
...
+        {
+            byte[] privateKey = { 0x01,0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+                0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01};
+            KeyPair keyPair = new(privateKey);
+            ECPoint pubkey = keyPair.PublicKey;
+
+            var engine = GetEngine();
+
+            engine.CheckWitness(pubkey.EncodePoint(true)).Should().BeFalse();
+            engine.CheckWitness(pubkey.EncodePoint(true)).Should().BeFalse();
```
Comment (vncoelho) [inline tests/Neo.UnitTests/SmartContract/UT_InteropService.cs:314]: Two checks?

Comment (Liaojinghui) [reply tests/Neo.UnitTests/SmartContract/UT_InteropService.cs:314]: can we just leave it this way for now? My bad, did not notice it at all.

Comment (vncoelho) [reply tests/Neo.UnitTests/SmartContract/UT_InteropService.cs:314]: Just fix in a new pr

**Pull request merged**